osquery/utils/windows/shelllnk.cpp

```diff
@@ -205,7 +206,12 @@ TargetInfo parseTargetInfo(const std::string& target_info) {
   target_lnk.path = osquery::join(build_path, "\\");
   target_lnk.mft_entry = file_entry.mft_entry;
   target_lnk.mft_sequence = file_entry.mft_sequence;
-  target_lnk.data = data;
+  // First two bytes equal the size of the whole target info data
+  std::string target_info_size = original_data.substr(0, 4);
+  target_info_size = swapEndianess(target_info_size);
+  int target_size = tryTo<int>(target_info_size, 16).takeOr(0) * 2;
```

Comment:
Can you clarify what these lines 210-212 are attempting to do:
```
  std::string target_info_size = original_data.substr(0, 4);
  target_info_size = swapEndianess(target_info_size);
  int target_size = tryTo<int>(target_info_size, 16).takeOr(0) * 2;
  ```
  
Isn't this just something similar to:
`size_t target_info_size = *(size_t*)original_data.c_str()`?

In this case, is the function `charToHexString` necessary?

Reply:
Shortcut files contain shellitems the function above parses the shellitem data. After parsing the shellitem data, the first 4 bytes contain the total size of the shellitem data. The lines above take the size of the shellitem data and removes the data and then continues parse the next part of the shortcut data.

Reply:
Ok, so, why is `swapEndianess` being called here? I am assuming `target_info_size` is little-endian, right? Or is `target_info_size` big-endian?

Reply:
Since its a hex string the format is big-endian.
It needs to be converted to little endian to get the size